test/e2e/v3/plugin_cluster_test.go

```diff
@@ -116,7 +119,13 @@ var _ = Describe("kubebuilder", func() {
 				}
 
 				GenerateV3(kbc, "v1")
-				Run(kbc)
+
+				// only if running on Kubernetes >= 1.24 do we need to generate the ServiceAccount token Secret
```

Comment:
Auto-generated tokens should not be scraped for use outside pods on any version, they are only for use inside pods.

I'd recommend using TokenRequest to get a token (this works on all currently supported Kubernetes versions) or consistently creating a dedicated token secret, but not scraping auto-generated tokens on some versions

Reply:
@liggitt that is only for the tests it is not used by the end-users and only because we curl the pod to check if we are able to export the metrics in the endpoint.

Reply:
still, I'd avoid scraping auto-generated tokens everywhere... you never know what code will get copied into more important locations

Reply:
The token also does still get used in a pod that runs the `curl` command.

I agree that it may be a good idea to see if there is a better approach to doing this as a follow up (ideally could track discussion in an issue), but I think the scope of this PR is specifically to minimally impact the current way the tests behave while still allowing them to work on K8s 1.24

Reply:
Sure thing. Will update it now.

Reply:
Yes, we can fix the CI now and create an issue to fix that. We have more changes that we need to there anyway.

Reply:
Added TODO comments in commit https://github.com/kubernetes-sigs/kubebuilder/pull/2723/commits/dfa14858054ddcb077205474cbbc6ed42a8376c5

Resolving this, but feel free to unresolve if there are further changes needed. Thanks!